amacc.c

```diff
@@ -871,73 +871,61 @@ void stmt(int ctx)
         return;
+    case Int:
+    case Char:
     case Struct:
-        next();
-        if (tk == Id) {
-            if (!id->stype) id->stype = tnew++;
-            bt = id->stype;
-            next();
-        } else { 
-            bt = tnew++;
-        }
-        if (tk == '{') {
-            next();
-            if (members[bt]) fatal("duplicate structure definition");
-            i = 0;
-            while (tk != '}') {
-                int mbt = INT;
-                switch (tk) {
-                case Int: next(); break;
-                case Char: next(); mbt = CHAR; break;
-                case Struct:
-                    next(); 
-                    if (tk != Id) fatal("bad struct declaration");
-                    mbt = id->stype;
-                    next(); break;
+        switch (tk) {
+        case Struct:
+                next();
+                if (tk == Id) {
+                    if (!id->stype) id->stype = tnew++;
+                    bt = id->stype;
+                    next();
+                } else { 
+                    bt = tnew++;
                 }
-                while (tk != ';') {
-                    ty = mbt;
-                    while (tk == Mul) { next(); ty += PTR; }
-                    if (tk != Id) fatal("bad struct member definition");
-                    m = malloc(sizeof(struct member_s));
-                    m->id = id;
-                    m->offset = i;
-                    m->type = ty;
-                    m->next = members[bt];
-                    members[bt] = m;
-                    i += (ty >= PTR) ? sizeof(int) : tsize[ty];
-                    i = (i + 3) & -4;
+                if (tk == '{') {
+                    next();
+                    if (members[bt]) fatal("duplicate structure definition");
+                    i = 0;
+                    while (tk != '}') {
+                        int mbt = INT;
+                        switch (tk) {
+                        case Int: next(); break;
+                        case Char: next(); mbt = CHAR; break;
+                        case Struct:
+                            next(); 
+                            if (tk != Id) fatal("bad struct declaration");
+                            mbt = id->stype;
+                            next(); break;
+                        }
+                        while (tk != ';') {
+                            ty = mbt;
+                            while (tk == Mul) { next(); ty += PTR; }
+                            if (tk != Id) fatal("bad struct member definition");
+                            m = malloc(sizeof(struct member_s));
+                            m->id = id;
+                            m->offset = i;
+                            m->type = ty;
+                            m->next = members[bt];
+                            members[bt] = m;
+                            i += (ty >= PTR) ? sizeof(int) : tsize[ty];
+                            i = (i + 3) & -4;
```

Comment:
I'm wondering if we shall prepare helper functions for alignment.

Reply:
Mmm. But autoformat tool will increase the code line,won't it?

Reply:
After code cleanup, I would like to enforce `clang-format` for consistent style. Sooner or later.